core/trino-main/src/main/java/io/trino/sql/planner/LogicalPlanner.java

```diff
@@ -272,6 +272,7 @@ public Plan plan(Analysis analysis, Stage stage, boolean collectPlanStatistics)
 
     private PlanOptimizer.Context createOptimizerContext()
     {
+        ExpressionInterpreter interpreter = new ExpressionInterpreter(plannerContext, session);
```

Comment:
`createOptimizerContext` (method name) looks like it's a context for an optimizer
also, the interface name `PlanOptimizer.Context` suggests that

Here, the intention is to have a long-lived `ExpressionInterpreter` which will live throughout the whole query optimization process. 

Let's create `ExpressionInterpreter` in the caller of this method, and pass it here as an argument.
Also, add a code comment explaining why it matters.

Reply:
done